src/bindings/python/src/pyopenvino/graph/model.cpp

```diff
@@ -327,19 +327,44 @@ void regclass_graph_Model(py::module m) {
     model.def(
         "reshape",
         [](ov::Model& self, const py::list& partial_shape) {
-            ov::PartialShape new_shape(Common::partial_shape_from_list(partial_shape));
-            py::gil_scoped_release release;
-            self.reshape(new_shape);
+            if (py::isinstance<py::list>(partial_shape[0])) {
+                std::map<ov::Output<ov::Node>, ov::PartialShape> shapes_map;
+
```

Comment:
should we check a number of shapes match to a number of inputs?

Reply:
Adding this check too, should we throw an error message if it's not matching? what error message do you suggest?

Reply:
Please cover this in this PR

Reply:
Yes I'll add it in this PR only

Reply:
@rkazants this check is already there, when the number of shapes don't match number of inputs it throws this error:

`FAILED tests/test_runtime/test_model.py::test_reshape_with_list_of_shapes - AssertionError: assert 4 == 3`

when input shape is 4 and in shapes list there are 3

Reply:
May be add such a test with failure?
https://docs.pytest.org/en/7.1.x/how-to/assert.html#assertions-about-expected-exceptions
And you may use parametrization like [here](https://github.com/openvinotoolkit/openvino/blob/master/src/bindings/python/tests/test_runtime/test_model.py#L47)

Reply:
@akuporos have added this now